### PR TITLE
Added script to support selenium grid

### PR DESCRIPTION
**Pull request by mdsapon**
adding grid url from either commandline or from sentinel.yml will run on grid

### CODE REVIEW

File: src/main/java/com/dougnoel/sentinel/webdrivers/WebDriverFactory.java [hunk 1]
```diff
@@ -92,8 +92,15 @@ public static WebDriver instantiateWebDriver() {
         var downloadDirectory = Configuration.toString("download");
         if (downloadDirectory != null)
             DownloadManager.setDownloadDirectory(downloadDirectory);
-
-        String browser = Configuration.browser();
+
```
Comment (dougnoel) [inline src/main/java/com/dougnoel/sentinel/webdrivers/WebDriverFactory.java:95]: Looks great.

File: src/main/java/com/dougnoel/sentinel/webdrivers/GridWebDriverFactory.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import com.dougnoel.sentinel.exceptions.MalformedURLException;
+
+public class GridWebDriverFactory {
+	
+	private GridWebDriverFactory() {
+		
+	}
+	
+	protected static WebDriver createGridDriver(String browser, String gridUrl) {
+    	var capability = new DesiredCapabilities();
```
Comment (dougnoel) [inline src/main/java/com/dougnoel/sentinel/webdrivers/GridWebDriverFactory.java:17]: This should all be indented.

File: src/main/java/com/dougnoel/sentinel/webdrivers/GridWebDriverFactory.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.remote.DesiredCapabilities;
+import org.openqa.selenium.remote.RemoteWebDriver;
+import com.dougnoel.sentinel.configurations.Configuration;
+import com.dougnoel.sentinel.exceptions.MalformedURLException;
+
+public class GridWebDriverFactory {
+	
+	private GridWebDriverFactory() {
+
```
Comment (dougnoel) [inline src/main/java/com/dougnoel/sentinel/webdrivers/GridWebDriverFactory.java:13]: Don't need the blank space here.

File: src/main/java/com/dougnoel/sentinel/webdrivers/GridWebDriverFactory.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+	
+	protected static WebDriver createGridDriver(String browser, String gridUrl) {
+    	var capability = new DesiredCapabilities();
+    	capability.setCapability("browserName", browser);
+        var browserVersion = Configuration.toString("browserVersion");
+    	if (browserVersion != null) 
+    	    capability.setCapability("browserVersion", browserVersion);
+    	
+    	URL url;
+        try {
```
Comment (dougnoel) [inline src/main/java/com/dougnoel/sentinel/webdrivers/GridWebDriverFactory.java:24]: This try should be on the same level as the URL line above.

**Review by dougnoel — CHANGES_REQUESTED** [review]

File: src/main/java/com/dougnoel/sentinel/webdrivers/GridWebDriverFactory.java (hunk 4 above)
Comment (mdsapon) [reply src/main/java/com/dougnoel/sentinel/webdrivers/GridWebDriverFactory.java:24]: That's weird, my eclipse shows it's in the same line

File: src/test/java/com/dougnoel/sentinel/webdrivers/WebDriverFactoryTest.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+		assertNull(driver);
+	}
+	
+	@Test(expected = org.openqa.selenium.remote.UnreachableBrowserException.class)
+	public void createGridShouldSucessTest() {
+		var browser = "chrome";
+		var gridUrl = "http://hub.technologynursery.org:4444/wd/hub";
+		var driver = GridWebDriverFactory.createGridDriver(browser, gridUrl);
+		 
+		assertNotNull(driver);
```
Comment (dougnoel) [inline src/test/java/com/dougnoel/sentinel/webdrivers/WebDriverFactoryTest.java:24]: This assertion is not giving us what we need. If you look at the code coverage, the URL code isn't being exercised.

File: src/test/java/com/dougnoel/sentinel/webdrivers/WebDriverFactoryTest.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+		var gridUrl = "http://gridrul.com";
+		var driver = GridWebDriverFactory.createGridDriver(browser, gridUrl);
+		 
+		assertNull(driver);
+	}
+	
+	@Test(expected = org.openqa.selenium.remote.UnreachableBrowserException.class)
+	public void createGridShouldSucessTest() {
+		var browser = "chrome";
+		var gridUrl = "http://hub.technologynursery.org:4444/wd/hub";
```
Comment (dougnoel) [inline src/test/java/com/dougnoel/sentinel/webdrivers/WebDriverFactoryTest.java:21]: You need a browser version set here for that code to be executed. you can do that with:

`System.setProperty("browserVersion","1234567");`

File: src/test/java/com/dougnoel/sentinel/webdrivers/WebDriverFactoryTest.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
+package com.dougnoel.sentinel.webdrivers;
+
+import static org.junit.Assert.*;
+
+import org.junit.Test;
+
+public class WebDriverFactoryTest {
+
+	@Test(expected = org.openqa.selenium.remote.UnreachableBrowserException.class)
```
Comment (dougnoel) [inline src/test/java/com/dougnoel/sentinel/webdrivers/WebDriverFactoryTest.java:9]: This is a good start, but you have some paths that aren't covered:
1.) Having a valid browserVersion (see below on how to to that.)
2.) Having a malformed URL and expecting an error.
3.) Having a driver returned.

You can see this in the code coverage report. I'd like to see 100% code coverage on this new file.

**Review by dougnoel — CHANGES_REQUESTED** [review]

**Review by dougnoel — APPROVED** [review]

**Pull request merged**